lib/client-assets.php

```diff
@@ -799,6 +799,41 @@ function gutenberg_capture_code_editor_settings( $settings ) {
 	$gutenberg_captured_code_editor_settings = $settings;
 	return false;
 }
+/**
+ * Retrieve a stored autosave that is newer than the post save.
+ *
+ * Deletes autosaves that are older than the post save.
+ *
+ * @param  WP_Post $post Post object.
+ * @return WP_Post|boolean The post autosave. False if none found.
+ */
+function get_autosave_newer_than_post_save( $post ) {
+	// Add autosave data if it is newer and changed.
+	$autosave      = wp_get_post_autosave( $post->ID );
+	$show_autosave = false;
+
+	// Check if we have an autosave that is newer than the post and different from the current post.
+	if (
+		$autosave &&
+		mysql2date( 'U', $autosave->post_modified_gmt, false ) > mysql2date( 'U', $post->post_modified_gmt, false )
+	) {
+		// Iterate thru revisioned fields checking for any changes.
+		foreach ( _wp_post_revision_fields( $post ) as $autosave_field => $_autosave_field ) {
+			if ( normalize_whitespace( $autosave->$autosave_field ) != normalize_whitespace( $post->$autosave_field ) ) {
+				$show_autosave = true;
+				break;
+			}
+		}
+	}
+
+	// If this autosave isn't newer and different from the current post, remove.
+	if ( $autosave && ! $show_autosave ) {
```

Comment:
This logic is different than the one in edit-form-advanced.php. Why would you want to delete older autosaves here?

Reply:
Trying to match this: https://core.trac.wordpress.org/browser/tags/4.9.5/src/wp-admin/edit-form-advanced.php#L217, possibly this is different.

essentially when the editor loads if the autosave isn't going to be shown (because its older) than the published post) delete it.

Reply:
The whole bit of code from there:
```
if ( $autosave && mysql2date( 'U', $autosave->post_modified_gmt, false ) > mysql2date( 'U', $post->post_modified_gmt, false ) ) {
  foreach ( _wp_post_revision_fields( $post ) as $autosave_field => $_autosave_field ) {
    if ( normalize_whitespace( $autosave->$autosave_field ) != normalize_whitespace( $post->$autosave_field ) ) {
      $notice = sprintf( __( 'There is an autosave of this post that is more recent than the version below. <a href="%s">View the autosave</a>' ), get_edit_post_link( $autosave->ID ) );
      break;
    }
  }
  // If this autosave isn't different from the current post, begone.
  if ( ! $notice )
    wp_delete_post_revision( $autosave->ID );

  unset($autosave_field, $_autosave_field);
}
```

Note how the `wp_delete_post_revision( $autosave->ID );` is inside the conditional that checks if the autosave is newer than the post, and depends on all autosave fields being equal to all post fields.

As mentioned above, this is an old bit of code that is not needed any more. Can keep it if needed but the logic has to be exactly the same as in edit-form-advanced.php.

Reply:
Ok, thanks for clarifying. I'll bring the existing logic over exactly. I think it is worth deleting stale revisions if only for cleanup purposes.